### PR TITLE
./run_tests changes & small bug fixes

### PR DESCRIPTION
**Pull request by gpotter2**
I often find myself running `./run_tests` because finding the correct tox environment is annoying, and because I often need too many `-K` options to make sure it runs properly on my machine. This PR fixes that

- Calling `./run_tests` with no arguments will now automatically call `tox -e ... -K manufdb -K wireshark...` with the correct environment. (NON ROOT tests only because we want it to be fast-ish). We re-use the `./config/ci/test.sh` script for that.
- this should also be the recommanded way to test scapy for package maintainers, as it's very quick and still gives a good idea of whether scapy works or not.

**Moved to https://github.com/secdev/scapy/pull/2933**
- ~~Bug fix: UTscapy would exit on 0 (success) if an invalid line was found in a `.uts` file.~~
- ~~Bug fix: two misplaced leftovers in the uts files~~
- ~~Bug fix: BGP metaclasses were called in the wrong order~~

### CODE REVIEW

**Review by guedou — COMMENTED** [review]
First of all, I think that the changes to the BGP layer and UTScapy must be in two different commits. They look like issues that must be fixed independently of this PR, and could be accepted without any discussion.

The core of this PR raises some issues:
- it changes the current behavior, and mandates to install tox to start the unit tests
- it loads the can kernel module silently. I think that it should never be loaded automatically unlike in CI

I don't think that this in an improvement against running tox manually, but I will suggest that you move these fixes under a dedicated argument (i.e. `./test/run_tests -tox`) or a standalone script (i.e. `./test/run_tests_tox`). Either way, the can kernel module should not be loaded automatically.

**Comment by gpotter2** [comment]
I agree about the can module, but I don't understand how this changes the default behavior.
Currently, running `./run_tests` with no arguments does nothing.

**Comment by guedou** [comment]
I am so used to type `./test/run_tests ./test/regression.uts` that I though it was the default. Sorry about that =/

`./test/run_tests_tox` seems like a fair compromise. What do you think?

**Comment by gpotter2** [comment]
Sounds good

**Comment by gpotter2** [comment]
Actually that's a bummer since we would have to copy paste everything in `run_tests` (the Python2/3 detection).
This is why I added the code here in the first place. It's not great to have to maintain both.
The example you posted doesn't work (it's missing the `-t` flag).

File: doc/scapy/development.rst [hunk 1]
```diff
@@ -254,6 +254,15 @@ all Scapy unit tests automatically without any external dependency::
 
  tox -- -K vcan_socket -K tcpdump -K tshark -K nmap -K manufdb -K crypto
 
+.. note:: This will trigger the unit tests on all available Python versions
+  unless you specify a `-e` option. See below
+
+For your convenience, and for package maintainers, we provide a util that
+run tox on only a single (default python) environment, again with no external
```
Comment (guedou) [inline doc/scapy/development.rst:261]: s/python/Python/

File: test/run_tests [hunk 2]
```diff
@@ -20,4 +34,27 @@ then
...
+  # No arguments specified: use tox
+  # We use flags to disable tests that use external non tox-installed
+  # software.
+
+  # Check tox
+  tox --version >/dev/null 2>/dev/null
+  if [ ! $? -eq 0 ]
+  then
+    echo "ERROR: tox is not installed."
+    echo "You can still run ./run_tests with arguments: try ./run_tests -h"
```
Comment (guedou) [inline test/run_tests:49]: Could you provide a usage example?

**Comment by guedou** [comment]
I understand. Could you do something to make sure that the can kernel module is not loaded?

File: test/run_tests [hunk 3]
```diff
@@ -20,4 +34,27 @@ then
...
+    exit 1
+  fi
+
+  # Run tox
+  export UT_FLAGS="-K tcpdump -K manufdb -K wireshark -K ci_only"
+  export SIMPLE_TESTS="true"
+  PYVER=$(python3 -c "import sys; print('.'.join(sys.version.split('.')[:2]))")
```
Comment (guedou) [inline test/run_tests:56]: Why are you enforcing python3 here while the script understand $PYTHON to specify the version to use.

Comment (gpotter2) [reply test/run_tests]: Good catch

**Comment by gpotter2** [comment]
Thanks for the review, I've applied the suggetsions.

**PLEASE SQUASH ON MERGING**

**Comment by guedou** [comment]
I will make more sense to remove your change to bgp.py and UTScapy.py from this PR and only keep the commits related to the tox mode for ./run_tests. What do you think?
If you are OK, I will change the ownership of my commits in #2933 related to bgp.py and utscapy.py

**Comment by gpotter2** [comment]
Sounds good. I'll wait for your PR so that I can rebase on it.

**Comment by guedou** [comment]
@gpotter2 done!

**Comment by gpotter2** [comment]
I rebased this.

**Pull request merged**